### DISCUSSION TITLE
How to customize the time widget?

### DISCUSSION THREAD

**Discussion by cshaa** [elementary/wingpanel#458] [Q&A] [answered]
### Question

I would love to modify my wingpanel, so that the date-time widget shows time in a custom format. (Specifically, I'd love to add [decimal time](https://en.wikipedia.org/wiki/Decimal_time) there.) However, I couldn't find what part of the source code is responsible for rendering the time. Could you point me in the right direction, please? :)

**Comment by tintou** [comment, marked as answer]
I'm sorry but we don't and won't support this level of customization.

The code you're looking for is at https://github.com/elementary/wingpanel-indicator-datetime/blob/master/src/Widgets/PanelLabel.vala#L60 also note that:
 * [GLib.DateTime](https://valadoc.org/glib-2.0/GLib.DateTime.format.html) has no way to rendering into decimal time
 * We're updating the label every minute or second only which won't match your custom format.

**Comment by cshaa** [comment]
Thanks!
